### DISCUSSION TITLE
Program combination

### DISCUSSION THREAD

**Discussion by QZheng82** [Anura3D/Anura3D_OpenSource#94] [Q&A] [answered]
Can you please tell me if I want to apply the random field to Anura3d for uncertainty simulation, how should I do it properly? Thanks!

**Comment by murphy-john** [comment, marked as answer]
We currently don't have a built in function to apply a random field in Anura3D. One could write a subroutine to assign properties from a random field during the initialization of the material points easily enough. Whether it can be done properly and get a meaningful answer is up to the user to decide.

**Reply by QZheng82** [reply]
Thanks! But I would like to ask which part of the code is the material initialisation board in Anura3d, because there are so many boards and I can't quite find them easily.

**Reply by murphy-john** [reply]
Finding things in the code on your own is an important skill to learn. 

You can find things in the code by looking through the "Kernal.FOR" file. It's structured  in a way that will take you step-by-step through the startup, running the model (loadstep then time step or iteration), then outputs. 

In this case, look for where the MPs are initialized. Search the code (use Ctrl+F to search or find) for the subroutine. You may have to go multiple levels down before arriving at the location where the MPs (they are called "Particles" in the code) are initialized.

**Comment by 6cs4u** [comment]
Dear QZheng82
After your research, is it feasible to use Anura3D for uncertainty analysis?
